Aplib.Core/Tactic.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Aplib.Core
+{
+    /// <summary>
+    /// Represents the type of a tactic.
+    /// </summary>
+    public enum TacticType
+    {
+        Primitive,
+        FirstOf,
+        AnyOf,
+    }
+
+    /// <summary>
+    /// Represents a tactic in the Aplib.Core namespace.
+    /// </summary>
+    public class Tactic
+    {
+        /// <summary>
+        /// Gets or sets the parent tactic.
+        /// </summary>
+        private Tactic? _parent = null;
+
+        /// <summary>
+        /// Gets or sets the type of the tactic.
+        /// </summary>
+        public TacticType TacticType { get; set; }
+
+        /// <summary>
+        /// Gets or sets the sub-tactics of the tactic.
+        /// </summary>
+        private LinkedList<Tactic> _subTactics { get; set; }
+
+        /// <summary>
+        /// Gets or sets the guard of the tactic.
+        /// </summary>
+        protected Func<bool> Guard { get; set; } = () => true;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Tactic"/> class with the specified tactic type and sub-tactics.
+        /// </summary>
+        /// <param name="tacticType">The type of the tactic.</param>
+        /// <param name="subTactics">The sub-tactics of the tactic.</param>
+        public Tactic(TacticType tacticType, List<Tactic> subTactics)
+        {
+            TacticType = tacticType;
+            _subTactics = new();
+
+            foreach (Tactic tactic in subTactics)
+            {
+                tactic._parent = this;
+                _ = _subTactics.AddLast(tactic);
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Tactic"/> class with the specified tactic type, sub-tactics, and guard.
+        /// </summary>
+        /// <param name="tacticType">The type of the tactic.</param>
+        /// <param name="subTactics">The sub-tactics of the tactic.</param>
+        /// <param name="guard">The guard of the tactic.</param>
+        public Tactic(TacticType tacticType, List<Tactic> subTactics, Func<bool> guard)
+        {
+            TacticType = tacticType;
+            _subTactics = new();
+            Guard = guard;
+
+            foreach (Tactic tactic in subTactics)
+            {
+                tactic._parent = this;
+                _ = _subTactics.AddLast(tactic);
+            }
+        }
```

Comment:
Remove the duplicate code here by using a `this` constructor:

```suggestion
        public Tactic(TacticType tacticType, List<Tactic> subTactics, Func<bool> guard)
        	: this(tacticType, subTactics)
        {
            Guard = guard;
        }
```